### PR TITLE
Complete task 5

### PR DESCRIPTION
**Pull request by 1Klevi1**
- [x] System Testing finished

### CODE REVIEW

**Comment by 1Klevi1** [comment]
I also fixed some code in diff classes as well, u can check the pull request

File: Ferry.java [hunk 1]
```diff
@@ -34,7 +34,7 @@ public Island getDestinationIsland() {
     public String processPass(Pass pass, Island island1, Island island2) {
         if (pass.getLuxuryRating() < island2.getIslandRating()) {
             return "The pass has a lower luxury rating than the destination island";
-        } else if (!island1.canPassEnter()) {
+        } else if (!island2.canPassEnter()) {
```
Comment (JackAshwell11) [inline Ferry.java:39]: Good bug fix

File: Resort.java [hunk 2]
```diff
@@ -183,7 +183,7 @@ public String travel(int pPassId, String ferCode) {
         Pass pass = getPass(pPassId);
         Ferry ferry = getFerry(ferCode);
         if (pass != null && ferry != null) {
-            return ferry.processPass(pass, getIsland(findPassLocation(pPassId)), ferry.getDestinationIsland());
+            return ferry.processPass(pass,ferry.getSourceIsland(), ferry.getDestinationIsland());
```
Comment (JackAshwell11) [inline Resort.java]: Have space between `pass,` and `ferry.getSourceIsland()`. Also idk why I didn't think of getting the source island like this

File: Tester.java [hunk 3]
```diff
@@ -8,10 +8,429 @@
  */
 public class Tester {
     private void doTest() {
-        FIRE fortunate = new Resort("Fortunate Islands", "temp location 2");
+        FIRE Wayward = new Resort("Wayward Islands Resort", "temp location 2");
```
Comment (JackAshwell11) [inline Tester.java]: You may want to change it to temp location 1, but it doesn't matter

File: Tester.java [hunk 4]
```diff
@@ -8,10 +8,429 @@
...
         Scanner enter = new Scanner(System.in);
+        //Testing option 1
+        System.out.println("""
+                Testing option 1,
```
Comment (JackAshwell11) [inline Tester.java]: We don't need the testing option 1 output here.

File: Tester.java [hunk 5]
```diff
@@ -8,10 +8,429 @@
...
+                ********************, ********************
+                Island ID Number: 4
+                Island Name: Aero
+                Island Rating: 1
+                Island capacity: 1
+                Pass List in Island:
+                []
+                ********************]
+                ********************
+                .Actual:
```
Comment (JackAshwell11) [inline Tester.java]: Remove the bullet point

File: Tester.java [hunk 6]
```diff
@@ -8,10 +8,429 @@
...
+                """ + "\n" + Wayward + "\n");
```
Comment (JackAshwell11) [inline Tester.java]: Remove the first newline here

File: Tester.java [hunk 7]
```diff
@@ -8,10 +8,429 @@
...
+//        Testing option 2
+        System.out.println("""
+                Testing option 2,
```
Comment (JackAshwell11) [inline Tester.java]: Don't need this testing option 2 output

File: Tester.java [hunk 8]
```diff
@@ -8,10 +8,429 @@
...
+                Credits: 20
+                Journey Points: 0
+                ********************
+                ********************
+                Guest Name: Tel
+                Pass ID Number: 1008
+                Luxury Rating: 6
+                Credits: 30
+                Journey Points: 0
+                ******************** .Actual:
```
Comment (JackAshwell11) [inline Tester.java]: Have actual on a new line with no bullet point

File: Tester.java [hunk 9]
```diff
@@ -8,10 +8,429 @@
...
+                """+ "\n" + Wayward.getAllPassesOnAllIslands() + "\n");
```
Comment (JackAshwell11) [inline Tester.java]: Remove first newline

File: Tester.java [hunk 10]
```diff
@@ -8,10 +8,429 @@
...
+        //        Testing option 3 List all passes on one Island
```
Comment (JackAshwell11) [inline Tester.java]: Reformat the spaces and capital letters here

File: Tester.java [hunk 11]
```diff
@@ -8,10 +8,429 @@
...
+        System.out.println("Option 3, List all passes on one Island in this case, island name: (Base)");
```
Comment (JackAshwell11) [inline Tester.java]: This class is meant to be automatic. There shouldn't be any user input

File: Tester.java [hunk 12]
```diff
@@ -8,10 +8,429 @@
...
+        String islandName = enter.next();
+        System.out.println(
+                """
```
Comment (JackAshwell11) [inline Tester.java]: Move triple quotes onto previous line

File: Tester.java (hunk 9 above)
Comment (JackAshwell11) [inline Tester.java]: Add space between triple quotes and plus

File: Tester.java [hunk 13]
```diff
@@ -8,10 +8,429 @@
...
+                ********************
+                ********************
+                Guest Name: Tel
+                Pass ID Number: 1008
+                Luxury Rating: 6
+                Credits: 30
+                Journey Points: 0
+                ********************
+                .Actual:""" + "\n" + Wayward.getAllPassesOnIsland(islandName) + "\n");
+        //        Testing option 4-> Find location of a pass
```
Comment (JackAshwell11) [inline Tester.java]: Same as earlier, remove the spaces and check capitals

File: Tester.java [hunk 14]
```diff
@@ -8,10 +8,429 @@
...
+        System.out.println("(True version option 4) Enter pass id:(1006)");
```
Comment (JackAshwell11) [inline Tester.java]: Redo everything from this point onwards. There should be no inputting in this class

File: Tester.java [hunk 15]
```diff
@@ -8,10 +8,429 @@
...
+                Expected:
+                ********************
+                Guest Name: May
+                Pass ID Number: 1001
+                Luxury Rating: 3
+                Credits: 16
+                Journey Points: 0
+                ********************
+                Actual:"""+"\n"+Wayward.viewAPass(cId4));
     }
```
Comment (JackAshwell11) [inline Tester.java:443]: Add the space back after this line

File: Tester.java [hunk 16]
```diff
@@ -8,10 +8,429 @@
  */
```
Comment (JackAshwell11) [inline Tester.java:6]: Add alot more comments and print outputs explaining what a specific test does. Thats one of the bullet points for task 5

**Review by JackAshwell11 — CHANGES_REQUESTED** [review]

File: Ferry.java (hunk 1 above)
Comment (1Klevi1) [reply Ferry.java:39]: yea that's what  I like when you work in pairs, the reviewing part

**Comment by 1Klevi1** [comment]
how is it?

File: Tester.java [hunk 17]
```diff
@@ -10,177 +11,15 @@ public class Tester {
     private void doTest() {
         FIRE Wayward = new Resort("Wayward Islands Resort", "temp location 2");
```
Comment (JackAshwell11) [inline Tester.java]: Does the variable have to be uppercase?

File: Tester.java [hunk 18]
```diff
@@ -10,177 +11,15 @@ public class Tester {
...
         Scanner enter = new Scanner(System.in);
```
Comment (JackAshwell11) [inline Tester.java]: Its best if this was deleted now

File: Tester.java [hunk 19]
```diff
@@ -10,177 +11,15 @@ public class Tester {
...
-                Pass ID Number: 1008
-                Luxury Rating: 6
-                Credits: 30
-                Journey Points: 0
-                ******************** .Actual:
-                """+ "\n" + Wayward.getAllPassesOnAllIslands() + "\n");
-        //        Testing option 3 List all passes on one Island
-        System.out.println("Option 3, List all passes on one Island in this case, island name: (Base)");
-        String islandName = enter.next();
+//      Testing option 3 List all passes on one Island
```
Comment (JackAshwell11) [inline Tester.java]: Format this comment correctly

Comment (JackAshwell11) [reply Tester.java]: Where's the tests for option 1 and 2?

File: Tester.java [hunk 20]
```diff
@@ -10,177 +11,15 @@ public class Tester {
...
+        System.out.println("Option 3, List all passes on one island in this case, island name: (Base)\n");
+        System.out.println(
+               """
+               This option will display all of the details of current passes that are currently
+               residing on an island. In order for this option to function properly,
+               you must specify one of the five islands that are: Base, Yorkie, Bounty, Twirl, and Aero.
+               """);
+        String islandName = "Base";
```
Comment (JackAshwell11) [inline Tester.java]: This is only used in one place so it doesn't need to be a variable

File: Tester.java [hunk 21]
```diff
@@ -247,91 +86,110 @@ private void doTest() {
                 Credits: 30
                 Journey Points: 0
                 ********************
-                .Actual:""" + "\n" + Wayward.getAllPassesOnIsland(islandName) + "\n");
-        //        Testing option 4-> Find location of a pass
-        System.out.println("(True version option 4) Enter pass id:(1006)");
-        int trav = enter.nextInt();
+                Actual:
+                """ + Wayward.getAllPassesOnIsland(islandName) + "\n");
+//      Testing option 4-> Find location of a pass
```
Comment (JackAshwell11) [inline Tester.java]: Format this comment and space out each option test so its easier to read

File: Tester.java [hunk 22]
```diff
@@ -247,91 +86,110 @@ private void doTest() {
...
+        System.out.println("(True version option 4) Pass id:(1006)\n");
```
Comment (JackAshwell11) [inline Tester.java]: I'd have this string as `Option 4 (valid) - Pass ID: 1006`

File: Tester.java [hunk 23]
```diff
@@ -247,91 +86,110 @@ private void doTest() {
...
+        System.out.println(
+               """
+               If you provide the right pass id, the software will return the current
+               island where a pass is located. If you enter the incorrect pass id,
+               the application will indicate that there is no such pass.
+               """);
+        int trav = 1006;
```
Comment (JackAshwell11) [inline Tester.java]: Combine this into the line below

File: Tester.java [hunk 24]
```diff
@@ -247,91 +86,110 @@ private void doTest() {
...
             System.out.println("""
                     Expected: Base
                     Actual:"""+ww+"\n");
-        } else {
-            System.out.println("No such pass"+"\n");
-        }
-        System.out.println("(False version option 4) Enter pass id:(111)");
-        int travel = enter.nextInt();
+        } else {System.out.println("No such pass"+"\n");}
+        System.out.println("(False version option 4) Pass id:(111)");
```
Comment (JackAshwell11) [inline Tester.java]: I'd have this string as `Option 4 (invalid) - Pass ID: 111`

File: Tester.java [hunk 25]
```diff
@@ -429,9 +292,10 @@ private void doTest() {
                 Credits: 16
                 Journey Points: 0
                 ********************
-                Actual:"""+"\n"+Wayward.viewAPass(cId4));
+                Actual:
+                """ + Wayward.viewAPass(cId4));
```
Comment (JackAshwell11) [inline Tester.java]: Just alot of the same comments. The formatting is off, make sure it's all spaced nicely and easy to read.

Also simplify some stuff as you have extra variables when you don't need to.

And option 1 and 2 are missing.

File: Tester.java [hunk 26]
```diff
@@ -429,9 +292,10 @@ private void doTest() {
...
     }
-    // No need to change this
+//      No need to change this
```
Comment (JackAshwell11) [inline Tester.java]: Why was this changed? It doesn't need to be

File: Tester.java [hunk 27]
```diff
@@ -429,9 +292,10 @@ private void doTest() {
...
     public static void main(String[] args) {
         Tester xx = new Tester();
```
Comment (JackAshwell11) [inline Tester.java]: Would look better if the variable name was changed to `tester` instead of `xx`.

**Review by JackAshwell11 — CHANGES_REQUESTED** [review]

**Comment by 1Klevi1** [comment]
how is it ?

File: Tester.java [hunk 28]
```diff
@@ -4,22 +4,192 @@
 /**
  * Write a description of class Tester here.
  *
- * @author Jack
- * @version 20/11/2022
+ * @author Klevi
```
Comment (JackAshwell11) [inline Tester.java:4]: Remove the scanner import

File: Tester.java [hunk 29]
```diff
@@ -4,22 +4,192 @@
...
+ * @version 09/12/2022
  */
 public class Tester {
     private void doTest() {
-        FIRE Wayward = new Resort("Wayward Islands Resort", "temp location 2");
-        Scanner enter = new Scanner(System.in);
-//      Testing option 3 List all passes on one Island
+        FIRE wayward = new Resort("Wayward Islands Resort", "temp location 2");
+
+        //Testing option 1
```
Comment (JackAshwell11) [inline Tester.java]: Space between `//` and `Testing`

File: Tester.java [hunk 30]
```diff
@@ -4,22 +4,192 @@
...
- * @version 20/11/2022
+ * @author Klevi
+ * @version 09/12/2022
  */
 public class Tester {
     private void doTest() {
-        FIRE Wayward = new Resort("Wayward Islands Resort", "temp location 2");
-        Scanner enter = new Scanner(System.in);
-//      Testing option 3 List all passes on one Island
+        FIRE wayward = new Resort("Wayward Islands Resort", "temp location 2");
```
Comment (JackAshwell11) [inline Tester.java]: Change the location to something a bit more suiting. Like `The Caribbean` or `The Mediterranean`

File: Tester.java [hunk 31]
```diff
@@ -4,22 +4,192 @@
...
+
+        //Testing option 1
+        System.out.println("""
+                Testing option 1,
```
Comment (JackAshwell11) [inline Tester.java]: Describe what this test does

File: Tester.java [hunk 32]
```diff
@@ -4,22 +4,192 @@
...
+                Island Rating: 1
+                Island capacity: 1
+                Pass List in Island:
+                []
+                ********************]
+                ********************
+                Actual:
+                """ + wayward + "\n");
+
+        //Testing option 2
```
Comment (JackAshwell11) [inline Tester.java]: Space between `//` and `testing`

File: Tester.java [hunk 33]
```diff
@@ -4,22 +4,192 @@
...
+        System.out.println("""
+                Testing option 2,
```
Comment (JackAshwell11) [inline Tester.java]: Describe what this test does

File: Tester.java [hunk 34]
```diff
@@ -4,22 +4,192 @@
...
+                Credits: 20
+                Journey Points: 0
+                ********************
+                ********************
+                Guest Name: Tel
+                Pass ID Number: 1008
+                Luxury Rating: 6
+                Credits: 30
+                Journey Points: 0
+                ********************
```
Comment (JackAshwell11) [inline Tester.java]: Theres an extra space here at the end

File: Tester.java [hunk 35]
```diff
@@ -4,22 +4,192 @@
...
+                Guest Name: Tel
+                Pass ID Number: 1008
+                Luxury Rating: 6
+                Credits: 30
+                Journey Points: 0
+                ******************** 
+                Actual:
+                """+  wayward.getAllPassesOnAllIslands() + "\n");
+
+        //Testing option 3 List all passes on one Island
```
Comment (JackAshwell11) [inline Tester.java]: Space between `//` and `testing` and remove the name. The test description is already printed

File: Tester.java [hunk 36]
```diff
@@ -4,22 +4,192 @@
...
         System.out.println("Option 3, List all passes on one island in this case, island name: (Base)\n");
```
Comment (JackAshwell11) [inline Tester.java]: This shouldnt be needed or should be merged into the output on the line below

File: Tester.java [hunk 37]
```diff
@@ -4,22 +4,192 @@
...
         System.out.println(
                """
```
Comment (JackAshwell11) [inline Tester.java]: Put this on the line above

File: Tester.java [hunk 38]
```diff
@@ -87,45 +257,42 @@ private void doTest() {
                 Journey Points: 0
                 ********************
                 Actual:
-                """ + Wayward.getAllPassesOnIsland(islandName) + "\n");
-//      Testing option 4-> Find location of a pass
-        System.out.println("(True version option 4) Pass id:(1006)\n");
+                """ + wayward.getAllPassesOnIsland("Base") + "\n");
```
Comment (JackAshwell11) [inline Tester.java]: How about a test for an island that doesn't exist?

File: Tester.java [hunk 39]
```diff
@@ -87,45 +257,42 @@ private void doTest() {
...
+
+        //Testing option 4-> Find location of a pass
```
Comment (JackAshwell11) [inline Tester.java]: Remove `-> Find location of a pass` and put the test description in the line below

File: Tester.java [hunk 40]
```diff
@@ -87,45 +257,42 @@ private void doTest() {
...
+        System.out.println("Option 4 (valid) - Pass ID: 1006\n");
         System.out.println(
                """
```
Comment (JackAshwell11) [inline Tester.java]: Put this on the line above

File: Tester.java [hunk 41]
```diff
@@ -87,45 +257,42 @@ private void doTest() {
...
                If you provide the right pass id, the software will return the current
                island where a pass is located. If you enter the incorrect pass id,
                the application will indicate that there is no such pass.
                """);
-        int trav = 1006;
-        String ww = Wayward.findPassLocation(trav);
+        String ww = wayward.findPassLocation(1006);
         if (ww != null) {
```
Comment (JackAshwell11) [inline Tester.java]: This if statement shouldn't be needed since the pass will always be found

File: Tester.java [hunk 42]
```diff
@@ -87,45 +257,42 @@ private void doTest() {
...
             System.out.println("""
                     Expected: Base
                     Actual:"""+ww+"\n");
         } else {System.out.println("No such pass"+"\n");}
-        System.out.println("(False version option 4) Pass id:(111)");
-        int travel = 111;
-        String www = Wayward.findPassLocation(travel);
+        System.out.println("Option 4 (invalid) - Pass ID: 111");
```
Comment (JackAshwell11) [inline Tester.java]: This line shouldn't be needed since the test description is on line 265

File: Tester.java [hunk 43]
```diff
@@ -87,45 +257,42 @@ private void doTest() {
...
+        String www = wayward.findPassLocation(111);
         System.out.println(Objects.requireNonNullElse(www, "No such pass" + "\n"));
```
Comment (JackAshwell11) [inline Tester.java]: This should just be `Expected: null. Actual: null`

File: Tester.java [hunk 44]
```diff
@@ -87,45 +257,42 @@ private void doTest() {
...
-                """ + Wayward.getAllPassesOnIsland(islandName) + "\n");
-//      Testing option 4-> Find location of a pass
-        System.out.println("(True version option 4) Pass id:(1006)\n");
+                """ + wayward.getAllPassesOnIsland("Base") + "\n");
+
+        //Testing option 4-> Find location of a pass
+        System.out.println("Option 4 (valid) - Pass ID: 1006\n");
         System.out.println(
                """
                If you provide the right pass id, the software will return the current
```
Comment (JackAshwell11) [inline Tester.java]: I'd reword this to `If you provide a valid pass ID, the pass’s current island will be returned. But if an invalid pass ID is provided, null will be returned.`

I'd even go as far as to describe the test inputs here:
```java
 System.out.println("""
               If you provide a valid pass ID, the pass’s current island will be returned. But if an invalid pass ID is provided, null will be returned.
               The tests:
                   1. Valid - Pass ID 1006.
                   2. Invalid - Pass ID 111.
               """);
```

File: Tester.java [hunk 45]
```diff
@@ -293,11 +437,11 @@ them is to use a ferry, so we are gonna use pass id (1001) because she has alrea
                 Journey Points: 0
                 ********************
                 Actual:
-                """ + Wayward.viewAPass(cId4));
+                """ + wayward.viewAPass(passId));
     }
-//      No need to change this
+        //No need to change this
```
Comment (JackAshwell11) [inline Tester.java]: Just have it like ` // No need to change this`

File: Tester.java [hunk 46]
```diff
@@ -87,45 +257,42 @@ private void doTest() {
...
-        System.out.println("(False version option 4) Pass id:(111)");
-        int travel = 111;
-        String www = Wayward.findPassLocation(travel);
+        System.out.println("Option 4 (invalid) - Pass ID: 111");
+        String www = wayward.findPassLocation(111);
         System.out.println(Objects.requireNonNullElse(www, "No such pass" + "\n"));
-//      Testing option 5-> Say if pass can move by ferry
-        System.out.println("(True Version, option 5) Pass id:(1000)\n");
+
+        //Testing option 5-> Say if pass can move by ferry
```
Comment (JackAshwell11) [inline Tester.java]: Remove `-> Say if pass can move by ferry`

File: Tester.java [hunk 47]
```diff
@@ -87,45 +257,42 @@ private void doTest() {
                 Journey Points: 0
                 ********************
                 Actual:
-                """ + Wayward.getAllPassesOnIsland(islandName) + "\n");
-//      Testing option 4-> Find location of a pass
-        System.out.println("(True version option 4) Pass id:(1006)\n");
+                """ + wayward.getAllPassesOnIsland("Base") + "\n");
+
+        //Testing option 4-> Find location of a pass
+        System.out.println("Option 4 (valid) - Pass ID: 1006\n");
```
Comment (JackAshwell11) [inline Tester.java]: This shouldn't be needed

File: Tester.java [hunk 48]
```diff
@@ -87,45 +257,42 @@ private void doTest() {
...
-        int travel = 111;
-        String www = Wayward.findPassLocation(travel);
+        System.out.println("Option 4 (invalid) - Pass ID: 111");
+        String www = wayward.findPassLocation(111);
         System.out.println(Objects.requireNonNullElse(www, "No such pass" + "\n"));
-//      Testing option 5-> Say if pass can move by ferry
-        System.out.println("(True Version, option 5) Pass id:(1000)\n");
+
+        //Testing option 5-> Say if pass can move by ferry
+        System.out.println("Option 5 (valid) - Pass id: 1000\n");
```
Comment (JackAshwell11) [inline Tester.java]: Not needed

File: Tester.java [hunk 49]
```diff
@@ -87,45 +257,42 @@ private void doTest() {
...
         System.out.println(
                """
```
Comment (JackAshwell11) [inline Tester.java]: Put on line above

File: Tester.java [hunk 50]
```diff
@@ -87,45 +257,42 @@ private void doTest() {
...
                For this option to operate correctly, you must enter the right pass id,
```
Comment (JackAshwell11) [inline Tester.java]: I'd reword this to `If a valid pass and ferry is provided and the conditions are met, true is returned. Otherwise, false is returned.` And with test inputs:
```java
        System.out.println("""
              If a valid pass and ferry is provided and the conditions are met, true is returned. Otherwise, false is 
              returned.
              The tests:
                  1. Valid - Pass ID 1000, Ferry Code ABC1.
                  2. Invalid - Pass ID 1002, Ferry Code ABC1.
                  3. More tests...
               """);
```

File: Tester.java [hunk 51]
```diff
@@ -87,45 +257,42 @@ private void doTest() {
...
-        System.out.println("Ferry code:(ABC1)");
-        String ferry1 = "ABC1";
-        System.out.println("Expected: false. Actual:"+Wayward.canTravel(trav2, ferry1) + "\n");
-//      Testing option 6-> Move a pass by ferry
-        System.out.println("(True version, option 6) Pass id:(1005)\n");
+        System.out.println("Ferry code: ABC1");
+        System.out.println("Expected: true. Actual:" + wayward.canTravel(1000, "ABC1") + "\n");
+        System.out.println("Option 5 (invalid) - Pass id: 1002\n");
+        System.out.println("Ferry code: ABC1");
+        System.out.println("Expected: false. Actual:" + wayward.canTravel(1002, "ABC1") + "\n");
```
Comment (JackAshwell11) [inline Tester.java]: Add more tests for invalid ferries and passes

File: Tester.java [hunk 52]
```diff
@@ -87,45 +257,42 @@ private void doTest() {
...
-        String ferry = "ABC1";
-        System.out.println("Expected: true. Actual:"+Wayward.canTravel(trav1, ferry) + "\n");
-        System.out.println("(False Version, option 5) Pass id:(1002)\n");
-        int trav2 = 1002;
-        System.out.println("Ferry code:(ABC1)");
-        String ferry1 = "ABC1";
-        System.out.println("Expected: false. Actual:"+Wayward.canTravel(trav2, ferry1) + "\n");
-//      Testing option 6-> Move a pass by ferry
-        System.out.println("(True version, option 6) Pass id:(1005)\n");
+        System.out.println("Ferry code: ABC1");
```
Comment (JackAshwell11) [inline Tester.java]: Use the code in the previous comment to detail the tests

File: Tester.java [hunk 53]
```diff
@@ -87,45 +257,42 @@ private void doTest() {
...
+        System.out.println("Expected: true. Actual:" + wayward.canTravel(1000, "ABC1") + "\n");
+        System.out.println("Option 5 (invalid) - Pass id: 1002\n");
+        System.out.println("Ferry code: ABC1");
+        System.out.println("Expected: false. Actual:" + wayward.canTravel(1002, "ABC1") + "\n");
+
+        //Testing option 6-> Move a pass by ferry
```
Comment (JackAshwell11) [inline Tester.java]: Remove `-> Move a pass by ferry`

File: Tester.java [hunk 54]
```diff
@@ -87,45 +257,42 @@ private void doTest() {
...
+        System.out.println("Option 6 (valid) - Pass id: 1005\n");
         System.out.println(
                """
```
Comment (JackAshwell11) [inline Tester.java]: Change this to
```java
        System.out.println("""
               If a valid pass and ferry code are provided, then the system will check the following conditions:
                   • The pass’s location is on the ferry’s source island.
                   • The pass has sufficient credits to travel.
                   • The pass’s luxury rating is the same or higher than the ferry’s destination island.
                   • The ferry’s destination island has sufficient capacity.
               If all the conditions are met, then the pass is allowed to travel. Otherwise, the reason is returned.
               The tests:
                   1. Valid - Pass ID 1005, Ferry Code ABC1.
                   2. Invalid - Pass ID 21, Ferry Code 21.
                   3. Invalid - Pass ID 1005, Ferry Code CDE3.
                   4. Invalid - Pass ID 1005, Ferry Code GHJ6
                   5. Invalid - Pass ID 1002, Ferry Code ABC1.
                   6. Invalid - Pass ID 1001, Ferry Code ABC1.
               """);
```

You can use html for the lists. I've done it in task 6

File: Tester.java [hunk 55]
```diff
@@ -135,61 +302,49 @@ private void doTest() {
                accept the pass. If everything said above is true, a message will appear stating,
                "The person can travel to the next island."
                """);
-        int trav3 = 1005;
-        System.out.println("Ferry code:(ABC1)");
-        String ferry2 = "ABC1";
-        System.out.println("Expected: The person can travel to the next island. Actual:"+Wayward.travel(trav3, ferry2) + "\n");
+        System.out.println("Ferry code: ABC1");
```
Comment (JackAshwell11) [inline Tester.java]: Most of these prints can be removed if you switch to the comment I added above

File: Tester.java [hunk 56]
```diff
@@ -135,61 +302,49 @@ private void doTest() {
...
-        int trav10 =1001;
-        System.out.println("Ferry code:(ABC1)");
-        String ferry9 = "ABC1";
-        System.out.println("Expected: The person is not on the source island. Actual:"+Wayward.travel(trav10, ferry9) + "\n");
-//      Testing option 7->View pass
+        System.out.println("Option 6 (invalid version 5) - The person is not on the source island for pass id: 1001");
+        System.out.println("Ferry code: ABC1");
+        System.out.println("Expected: The person is not on the source island. Actual:" + wayward.travel(1001, "ABC1") + "\n");
+
+        //Testing option 7->View pass
```
Comment (JackAshwell11) [inline Tester.java]: Remove ->View pass

File: Tester.java [hunk 57]
```diff
@@ -135,61 +302,49 @@ private void doTest() {
...
         System.out.println(
```
Comment (JackAshwell11) [inline Tester.java]: Change to
```java
        System.out.println("""
                If a valid pass ID is provided, the pass details are returned. Otherwise, the reason is returned.
                The tests:
                    1. Valid – Pass ID 1003 (before and after a state change).
                    2. Invalid - Pass ID (give an invalid pass ID).
```

File: Tester.java [hunk 58]
```diff
@@ -220,14 +372,13 @@ private void doTest() {
                 Journey Points: 1
                 ********************
                 Actual:
-                """ + Wayward.viewAPass(cId1) + "\n");
-//      Testing option 8->Top up credits
+                """ + wayward.viewAPass(1003) + "\n");
+
+        //Testing option 8->Top up credits
```
Comment (JackAshwell11) [inline Tester.java]: Remove `->Top up credits`

File: Tester.java [hunk 59]
```diff
@@ -220,14 +372,13 @@ private void doTest() {
...
         System.out.println("Option 8 requires us to provide the right pass id, followed by any amount of credits more than zero.");
-        System.out.println("Option 8, Please enter the id of the pass topping up their credits:(1003)");
-        int passId = 1003;
-        System.out.println("Please enter the number of credits to be added:(100)");
-        int credits = 100;
-        Wayward.topUpCredits(passId, credits);
+        System.out.println("Option 8, Please enter the id of the pass topping up their credits: 1003");
```
Comment (JackAshwell11) [inline Tester.java]: Add a check in `Pass.addCredits()` to only change credits if points is negative. So `if (credits > 0) {`

File: Tester.java [hunk 60]
```diff
@@ -220,14 +372,13 @@ private void doTest() {
                 Journey Points: 1
                 ********************
                 Actual:
-                """ + Wayward.viewAPass(cId1) + "\n");
-//      Testing option 8->Top up credits
+                """ + wayward.viewAPass(1003) + "\n");
+
+        //Testing option 8->Top up credits
         System.out.println("Option 8 requires us to provide the right pass id, followed by any amount of credits more than zero.");
```
Comment (JackAshwell11) [inline Tester.java]: Change to
```java
        System.out.println("""
                If a valid pass ID is provided along with a positive credit number, the pass’s credit amount is increased.
                Otherwise, there is no change in state.
                The tests:
                    1. Valid - Pass ID 1003, Credits 100.
                    2. Invalid - (add test for invalid pass id and valid credits).
                    3. Invalid - (add test for valid pass and invalid credits).
```

File: Tester.java [hunk 61]
```diff
@@ -239,31 +390,26 @@ private void doTest() {
                 Journey Points: 1
                 ********************
                 Actual:
-                """ + Wayward.viewAPass(passId) + "\n");
-//      Testing option 9->Convert points to credits
+                """ + wayward.viewAPass(1003) + "\n");
+
+        //Testing option 9->Convert points to credits
```
Comment (JackAshwell11) [inline Tester.java]: Remove `->Convert points to credits`

File: Tester.java [hunk 62]
```diff
@@ -239,31 +390,26 @@ private void doTest() {
...
         System.out.println(
                """
```
Comment (JackAshwell11) [inline Tester.java]: Change to:
```java
        System.out.println("""
                If a valid pass ID is provided, then their journey points are converted to credits. However, to conduct this test, we must get a pass with 5 journey points which can only be achieved through using the ferry.
                The tests:
                    1. Valid - Pass ID 1001 (before and after convert).
                    2. Invalid - (test a pass that does not have >=5 journey points).
```

File: Tester.java [hunk 63]
```diff
@@ -293,11 +437,11 @@ them is to use a ferry, so we are gonna use pass id (1001) because she has alrea
...
     }
-//      No need to change this
+        //No need to change this
     public static void main(String[] args) {
-        Tester xx = new Tester();
-        xx.doTest();
+        Tester tester = new Tester();
+        tester.doTest();
     }
 }
```
Comment (JackAshwell11) [inline Tester.java:450]: Pretty good, some formatting and detailing changes plus a few more tests and it should be done

**Review by JackAshwell11 — CHANGES_REQUESTED** [review]

**Comment by 1Klevi1** [comment]
yep done, what do u think?

File: Resort.java [hunk 64]
```diff
@@ -126,14 +126,14 @@ public int getIslandNumber(String isl) {
     public String getAllPassesOnIsland(String isl) {
         int targetInd = getIslandNumber(isl);
         if (targetInd == -1) {
-            return "";
-        } else {
-            String s = "";
+            return "The island doesn't exist";
+        } else{
```
Comment (JackAshwell11) [inline Resort.java]: Add space after `else`

**Review by JackAshwell11 — CHANGES_REQUESTED** [review]

**Review by JackAshwell11 — APPROVED** [review]
